### ISSUE TITLE
FIND/PART appears broken in Ren-C

### ISSUE THREAD

**Issue by hostilefork** [metaeducation/ren-c#874]
Let **s: "abcd"**.  In R3-Alpha and Red:

    >> find/part s "a" s
    == none

    >> find/part s "a" next s
    == "abcd"

    >> find/part s "ab" next s
    == "abcd"

    >> find/part s "bc" next s
    == none

In Ren-C:

    >> find/part s "a" s
    ;-- null

    >> find/part s "a" next s
    == "abcd"

    >> find/part s "ab" next s
    ;-- null // !!! This isn't right

     >> find/part s "bc" next s
     ;-- null

Rebol2 incidentally gives that same result, but for likely different reasons (odds are it wouldn't be the same bug?)

**Comment by hostilefork** [comment]
I think the intended interpretation of FIND/PART is to act as if the data being searched is limited to what would be seen e.g. by COPY/PART.  So it is effectively subsetting the input string for the purpose of the search, and answering appropriately.

Under this definition, Ren-C and Rebol2 are correct.  Again with **s: "abcd"**:

     >> copy/part s s
     == ""

     >> find/part s "a" s
     ;-- null

     >> copy/part s next s
     == "a"

     >> find/part s "a" next s
     == "abcd"

     >> find/part s "ab" next s
     ;-- null

In other words, you can't find "ab" in the "a" you would get, e.g. by `copy/part` of s to next s.  So you shouldn't be able to find it when you `find/part` of s to next s.

If this is the philosophy of the operation, then R3-Alpha and Red are the ones with the bug.
